document-store/src/main/java/org/hypertrace/core/documentstore/mongo/query/transformer/MongoSelectionsAddingTransformer.java

```diff
@@ -0,0 +1,30 @@
+package org.hypertrace.core.documentstore.mongo.query.transformer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.hypertrace.core.documentstore.expression.type.SelectingExpression;
+import org.hypertrace.core.documentstore.query.Query;
+import org.hypertrace.core.documentstore.query.Query.TransformedQueryBuilder;
+import org.hypertrace.core.documentstore.query.QueryTransformer;
+import org.hypertrace.core.documentstore.query.SelectionSpec;
+
+final class MongoSelectionsAddingTransformer implements QueryTransformer {
+  @Override
+  public Query transform(final Query query) {
+    List<SelectionSpec> newSpecs = new ArrayList<>();
+
+    for (SelectionSpec spec : query.getSelections()) {
+      String alias = spec.getAlias();
```

Comment:
Same question regarding alias being null/empty

Reply:
Yeah. Good catch. Need to make alias mandatory for AggregationExpression since we cannot work without them in MongoDB. Will handle it.